nodebuilder/tests/swamp/swamp.go

```diff
@@ -200,6 +200,9 @@ func (s *Swamp) DefaultTestConfig(tp node.Type) *nodebuilder.Config {
 
 	cfg.Core.IP = ip
 	cfg.Core.GRPCPort = port
+
+	// trigger flushes more often in tests.
+	cfg.Header.Store.WriteBatchSize = 1
```

Comment:
Instead of adding more logic & timeouts in tests it is simpler to force flushes more often.